src/content_main.js

```diff
@@ -171,42 +162,134 @@ async function storeRecentTimeSpent(recentTimeSpent) {
 }
 
 /***********************************************************
- * Validate Payment Pointer
+ * Track and Validate Payment Pointer
  ***********************************************************/
 
 /**
- * Check for a monetization meta tag on the website and verify that
- * the payment pointer is valid (resolves to a valid SPSP endpoint).
- * 
- * TODO: use enum to indicate no meta tag, meta tag + valid endpoint,
- * meta tag + invalid endpoint.
- * 
- * @return {Promise<{ isPaymentPointerValid: boolean, paymentPointer:string}>} 
- * isPaymentPointerValid is true if both monetization is present and the payment endpoint is valid.
- * paymentPointer is the paymentPointer if it is found in the monetization meta tag, otherwise null.
+ * getPaymentPointerFromPage gets the payment pointer as a string from the monetization meta tag
+ * on the current page. If there is no monetization meta tag on the page, null is returned.
+ * For more info on the monetization meta tag: https://webmonetization.org/docs/getting-started
+ *
+ * @return {string|null} The payment pointer on the current page, or null if no meta tag is present.
  */
-async function getAndValidatePaymentPointer() {
-	const monetizationMeta = document.querySelector('meta[name="monetization"]');
-	let paymentPointer = (monetizationMeta) ? monetizationMeta.content : null;
-	let isValid = false;
+function getPaymentPointerFromPage() {
+	const monetizationMetaTag = document.querySelector('meta[name="monetization"]');
+	if (monetizationMetaTag) {
+		return monetizationMetaTag.content;
+	} else {
+		return null;
+	}
+}
+
+/**
+ * trackPaymentPointer checks repeatedly (every NEW_PAYMENT_POINTER_CHECK_RATE_MS milliseconds)
+ * to see if a new payment pointer is added to the page. If a new payment pointer is added to the
+ * page, handleValidationAndSetExtensionIcon is called to
+ *	- ensure the payment pointer is validated,
+ *	- store the payment pointer in storage,
+ *	- and update the extension icon to the correct image.
```

Comment:
The description on `handleValidationAndSetExtensionIcon` would be better placed outside of this description (possibly inline with related code, or in `handleValidationAndSetExtensionIcon` itself), since this description should focus on describing `trackPaymentPointer`